Test/Tests/Network/DNS.Tests.ps1

```diff
@@ -23,63 +23,61 @@ Param (
 . $PSScriptRoot\..\..\Utils\ComputeNode\Initialize.ps1
 . $PSScriptRoot\..\..\Utils\DockerNetwork\DockerNetwork.ps1
 . $PSScriptRoot\..\..\Utils\ComputeNode\Configuration.ps1
-. $PSScriptRoot\..\..\Utils\DockerNetwork\DockerNetwork.ps1
-. $PSScriptRoot\..\..\Utils\ContrailAPI\DNSServerRepo.ps1
-. $PSScriptRoot\..\..\Utils\ContrailAPI\IPAMRepo.ps1
-. $PSScriptRoot\..\..\Utils\ContrailAPI\VirtualNetwork.ps1
+
+. $PSScriptRoot\..\..\Utils\ContrailAPI_New\ContrailAPI.ps1
+. $PSScriptRoot\..\..\Utils\TestCleanup\TestCleanup.ps1
 
 $ContrailProject = 'ci_tests_dns'
 
-$ContainersIDs = @("jolly-lumberjack","juniper-tree")
+$ContainersIDs = @('jolly-lumberjack', 'juniper-tree')
 
-$Subnet = [SubnetConfiguration]::new(
-    "10.0.5.0",
+$Subnet = [Subnet]::new(
+    '10.0.5.0',
     24,
-    "10.0.5.1",
-    "10.0.5.19",
-    "10.0.5.83"
+    '10.0.5.1',
+    '10.0.5.19',
+    '10.0.5.83'
 )
-$Network = [Network]::New("testnet12", $Subnet)
+$VirtualNetwork = [VirtualNetwork]::New('testnet_dns', $ContrailProject, $Subnet)
 
-$TenantDNSServerAddress = "10.0.5.80"
+$TenantDNSServerAddress = '10.0.5.80'
 
-$VirtualDNSServer = [DNSServer]::New("CreatedForTest")
+$VirtualDNSServer = [DNSServer]::New('CreatedForTest')
 
-$VirtualDNSrecords = @([DNSRecord]::New("vdnsrecord-nonetest", "1.1.1.1", "A", $VirtualDNSServer.GetFQName()),
-                       [DNSRecord]::New("vdnsrecord-defaulttest", "1.1.1.2", "A", $VirtualDNSServer.GetFQName()),
-                       [DNSRecord]::New("vdnsrecord-virtualtest", "1.1.1.3", "A", $VirtualDNSServer.GetFQName()),
-                       [DNSRecord]::New("vdnsrecord-tenanttest", "1.1.1.4", "A", $VirtualDNSServer.GetFQName())
-)
+$VirtualDNSrecords = @([DNSRecord]::New('vnone', $VirtualDNSServer.GetFQName(), 'vdnsrecord-nonetest', '1.1.1.1', 'A'),
+    [DNSRecord]::New('vdefa', $VirtualDNSServer.GetFQName(), 'vdnsrecord-defaulttest', '1.1.1.2', 'A'),
+    [DNSRecord]::New('vvirt', $VirtualDNSServer.GetFQName(), 'vdnsrecord-virtualtest', '1.1.1.3', 'A'),
+    [DNSRecord]::New('vtena', $VirtualDNSServer.GetFQName(), 'vdnsrecord-tenanttest', '1.1.1.4', 'A'))
 
-$DefaultDNSrecords = @([DNSRecord]::New("defaultrecord-nonetest.com", "3.3.3.1", "A", ""),
-                       [DNSRecord]::New("defaultrecord-defaulttest.com", "3.3.3.2", "A", ""),
-                       [DNSRecord]::New("defaultrecord-virtualtest.com", "3.3.3.3", "A", ""),
-                       [DNSRecord]::New("defaultrecord-tenanttest.com", "3.3.3.4", "A", "")
-)
+$DefaultDNSrecords = @([DNSRecord]::New('vnone', $null, 'defaultrecord-nonetest.com', '3.3.3.1', 'A'),
+    [DNSRecord]::New('vdefa', $null, 'defaultrecord-defaulttest.com', '3.3.3.2', 'A'),
+    [DNSRecord]::New('vvirt', $null, 'defaultrecord-virtualtest.com', '3.3.3.3', 'A'),
+    [DNSRecord]::New('vtena', $null, 'defaultrecord-tenanttest.com', '3.3.3.4', 'A'))
 
 # This function is used to generate command
 # that will be passed to docker exec.
 # $Hostname will be substituted.
```

Comment:
Please add a comment regarding deleted `else`. It might be tempting for someone to revert it, because it's a _quick win_ refactoring. Since we removed it by design, we could signal to people reading the code in the future that it might cause some troubles.